packages/react-components/src/components/TextArea/TextArea.tsx

```diff
@@ -0,0 +1,85 @@
+import { useState } from "react";
+import {
+  TextField as ReactAriaTextField,
+  TextFieldProps as ReactAriaTextFieldProps,
+  TextArea as ReactAriaTextArea,
+  Label,
+  FieldError,
+  Text,
+  ValidationResult,
+} from "react-aria-components";
+
+import "./TextArea.css";
+
+export interface TextAreaProps extends ReactAriaTextFieldProps {
+  /* Sets text label above text input field */
+  label?: string;
+  /* Sets optional description text below text input field */
+  description?: string;
+  /* Used for data validation and error handling */
+  errorMessage?: string | ((validation: ValidationResult) => string);
+}
+
+export default function TextArea({
+  label,
+  description,
+  errorMessage,
+  maxLength,
+  onChange,
+  value,
+  ...props
+}: TextAreaProps) {
+  // handler for character counter, displayed when a maxLength is set
+  const [charCount, setCharCount] = useState(value ? value.length : 0);
+
+  function handleChange(text: string) {
+    if (onChange) onChange(text);
+    setCharCount(text.length);
+  }
+
+  return (
+    <ReactAriaTextField className={`bcds-react-aria-TextArea`} {...props}>
+      {({ isRequired }) => (
+        <>
+          {label && (
+            <Label className="bcds-react-aria-TextArea--Label">
+              {label}
+              {isRequired && (
+                <span className="bcds-react-aria-TextArea--Label required">
+                  (required)
+                </span>
+              )}
+            </Label>
+          )}
+          <div className="bcds-react-aria-TextArea--Container">
+            <ReactAriaTextArea
+              className="bcds-react-aria-TextArea--Input"
+              value={value}
+              onChange={(e) => handleChange(e.target.value)}
+            />
+          </div>
+          {description || maxLength ? (
+            <div className="bcds-react-aria-TextArea--Description">
+              {description && (
+                <Text
+                  slot="description"
+                  className="bcds-react-aria-TextArea--Description text"
```

Comment:
This class name and the class name for the counter are hard to understand because we're using `bcds-react-aria-TextArea--Description` for the parent. Let's be very explicit and rename this to just `text` and write the selector like `bcds-react-aria-TextArea--Description > text`, or use `bcds-react-aria-TextArea--Description-Text` or something.

Reply:
Done in [49038e3](https://github.com/bcgov/design-system/pull/400/commits/49038e3898f87f9ed629a9bb1f67fcd48966eb3f)